static/js/stream_list.js

```diff
@@ -529,29 +540,143 @@ exports.clear_and_hide_search = function () {
 };
 
 function focus_stream_filter(e) {
+    if ($('#stream_filters li.narrow-filter.highlighted_stream').length === 0) {
+        // Highlight
+        var all_streams = $('#stream_filters li.narrow-filter');
+        exports.highlight_first(all_streams, 'highlighted_stream');
+    }
     e.stopPropagation();
 }
 
-function maybe_select_stream(e) {
-    if (e.keyCode === 13) {
-        // Enter key was pressed
+function focusout_stream_filter() {
+    // Undo highlighting
+    $('#stream_filters li.narrow-filter.highlighted_stream').removeClass('highlighted_stream');
+}
+
+function keydown_enter_key() {
+    // Is there at least one stream?
+    if ($('#stream_filters li.narrow-filter').length > 0) {
+
+        var selected_stream_id = $('#stream_filters li.narrow-filter.highlighted_stream')
```

Comment:
We usually don't do a blank line right after an `if` statement before starting to declare things.